x86_64-darwin build broken

## Describe the bug

https://hydra.nixos.org/build/290788151

```
../nix_api_expr.cc:38:18: error: aligned allocation function of type 'void *(std::size_t, std::align_val_t)' is only available on macOS 10.13 or newer
    void * p = ::operator new(
                 ^
../nix_api_expr.cc:38:18: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
1 error generated.
```

Seems to be introduced in 82a23d9b6b96bf08e7c28008fcc346c0bdb671be. @roberth 

## Steps To Reproduce

<!--
  Example:

  1. Clone this repository: ...
  2. Run `nix-... ...`
  3. Observe unexpected behaviour
-->

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->

## Metadata

<!-- Please insert the output of running `nix-env --version` below this line -->

## Additional context

<!-- Add any other context about the problem here. -->

## Checklist

<!-- make sure this issue is not redundant or obsolete -->

- [ ] checked [latest Nix manual] \([source])
- [ ] checked [open bug issues and pull requests] for possible duplicates

[latest Nix manual]: https://nixos.org/manual/nix/unstable/
[source]: https://github.com/NixOS/nix/tree/master/doc/manual/source
[open bug issues and pull requests]: https://github.com/NixOS/nix/labels/bug

---

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

--- a/flake.nix
+++ b/flake.nix
@@ -154,7 +154,6 @@
                 f = import ./packaging/components.nix {
                   inherit (final) lib;
                   inherit officialRelease;
-                  inherit stdenv;
                   pkgs = final;
                   src = self;
                 };

--- a/packaging/components.nix
+++ b/packaging/components.nix
@@ -2,7 +2,6 @@
   lib,
   pkgs,
   src,
-  stdenv,
   officialRelease,
 }:
 
@@ -11,6 +10,15 @@ scope:
 let
   inherit (scope)
     callPackage
+    ;
+  inherit
+    (scope.callPackage (
+      { stdenv }:
+      {
+        inherit stdenv;
+      }
+    ) { })
+    stdenv
     ;
   inherit (pkgs.buildPackages)
     meson

